src/java/org/apache/cassandra/cql3/statements/BatchStatement.java

```diff
@@ -235,7 +261,7 @@ public void validate() throws InvalidRequestException
             for (ModificationStatement stmt : statements)
             {
                 if (ksName != null && (!stmt.keyspace().equals(ksName) || !stmt.table().equals(cfName)))
-                    throw new InvalidRequestException("Batch with conditions cannot span multiple tables");
+                    throw new InvalidRequestException("Batch with conditions cannot span multiple tables: " + stmt.source.describe(true));
```

Comment:
Is there any coverage for these two cases in `BatchTest`? In general, feels like it would be good to have a test for the `ModificationStatement` case...